packages/core/test/acceptance/profiler_spec.ts

```diff
@@ -0,0 +1,241 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ProfilerEvent, setProfiler} from '@angular/core/src/render3/profiler';
+import {TestBed} from '@angular/core/testing';
+import {expect} from '@angular/core/testing/src/testing_internal';
+import {onlyInIvy} from '@angular/private/testing';
+
+import {Component, ErrorHandler, EventEmitter, OnInit, Output, ViewChild} from '../../src/core';
+
+
+onlyInIvy('Ivy-specific functionality').describe('profiler', () => {
+  class Profiler {
+    profile() {}
+  }
+
+  let profilerSpy: jasmine.Spy;
+
+  beforeEach(() => {
+    const profiler = new Profiler();
+    profilerSpy = spyOn(profiler, 'profile').and.callThrough();
+    setProfiler(profiler.profile);
+  });
+
+  describe('change detection hooks', () => {
+    it('should call the profiler for creation and change detection', () => {
+      @Component({selector: 'my-comp', template: '<button (click)="onClick()"></button>'})
+      class MyComponent {
+        onClick() {}
+      }
+
+      TestBed.configureTestingModule({declarations: [MyComponent]});
+      const myComp = TestBed.createComponent(MyComponent);
+
+      expect(profilerSpy).toHaveBeenCalled();
+      const [creationStart, creationEnd] = profilerSpy.calls.all();
+
+      expect(creationStart.args[0]).toBe(ProfilerEvent.CreationStart);
+      expect(creationStart.args[1]).toBe(myComp.componentInstance);
+
+      expect(creationEnd.args[0]).toBe(ProfilerEvent.CreationEnd);
+      expect(creationEnd.args[1]).toBe(myComp.componentInstance);
+
+      myComp.detectChanges();
+
+      const [changeDetectionStart, changeDetectionEnd] = profilerSpy.calls.all().slice(-2);
+
+      expect(changeDetectionStart.args[0]).toBe(ProfilerEvent.ChangeDetectionStart);
+      expect(changeDetectionStart.args[1]).toBe(myComp.componentInstance);
+
+      expect(changeDetectionEnd.args[0]).toBe(ProfilerEvent.ChangeDetectionEnd);
+      expect(changeDetectionEnd.args[1]).toBe(myComp.componentInstance);
+    });
+
+    it('should invoke the profiler when the template throws', () => {
+      @Component({selector: 'my-comp', template: '{{ throw() }}'})
+      class MyComponent {
+        throw() {
+          throw new Error();
+        }
+      }
+
+      TestBed.configureTestingModule({declarations: [MyComponent]});
+
+      let myComp: MyComponent;
+      expect(() => {
+        const fixture = TestBed.createComponent(MyComponent);
+        myComp = fixture.componentInstance;
+        fixture.detectChanges();
+      }).toThrow();
+
+      expect(profilerSpy).toHaveBeenCalled();
+      const [creationStart, creationEnd] = profilerSpy.calls.all();
+
+      expect(creationStart.args[0]).toBe(ProfilerEvent.CreationStart);
+      expect(creationStart.args[1]).toBe(myComp!);
```

Comment:
You could probably use some method like `toHaveBeenCalledWith` and `toHaveBeenCalledTimes` here and also use `argsFor`, for ex:

```suggestion
      profilerSpy.calls.argsFor(0).toBe([ProfilerEvent.CreationStart, myComp]);
      profilerSpy.calls.argsFor(1).toBe([ProfilerEvent.CreationEnd, myComp]);
```